File: android/arms/src/main/java/com/harvbot/arms/HarvbotArmFactory.java
package com.harvbot.arms;

import android.hardware.usb.UsbDevice;
import android.hardware.usb.UsbManager;

import java.io.IOException;

import static com.harvbot.arms.HarvbotArmTypes.*;

public class HarvbotArmFactory
{
    public static HarvbotArmBase getInstance(HarvbotArmTypes armType, UsbManager usbManager, UsbDevice device) throws IOException {
        switch(armType)
        {
            case Type1:
                return new HarvbotArm1(usbManager, device);
            default:
                throw new IllegalArgumentException("Invalid Harvbot Arm type");
        }
    }
}

File: .net/Harvbot.Arms.Driver/HarvbotArmFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Harvbot.Arms.Driver
{
    /// <summary>
    /// Represents Harvbot Arm factory.
    /// </summary>
    public static class HarvbotArmFactory
    {
        /// <summary>
        /// Gets the Harvbot Arm instance.
        /// </summary>
        /// <param name="comNum">The COM number.</param>
        /// <param name="type">The Arm type.</param>
        /// <returns>The Arm instance.</returns>
        public static HarvbotArmBase GetInstance(string comNum, HarvbotArmTypes type)
        {
            switch (type)
            {
                case HarvbotArmTypes.Type1:
                    {
                        return new HarvbotArm1(comNum);
                    }
                default:
                    {
                        throw new NotSupportedException("Specified arm type is not supported.");
                    }
            }
        }
    }
}
